vehicle/mkv/software/air_control/air.c

```diff
@@ -1,499 +1,435 @@
+        set_fault(AIR_FAULT_CAN_MC_TIMEOUT);
+        rc = 1;
+        goto bail;
+    }
 
-void resetTimer1(void) {
-		cli();
-		TCNT1H = 0x00; // write timer count to 0, high byte must be written first per datasheet
-		TCNT1L = 0x00;
-		sei();
-		timer1OverflowCount = 0x00;
-}
+    if (mc_voltage > MOTOR_CONTROLLER_THRESHOLD_LOW_dV) {
+        set_fault(AIR_FAULT_MOTOR_CONTROLLER_VOLTAGE);
+        rc = 1;
+        goto bail;
+    }
 
-void setOutputs(void) {
-		//Sets these pins at outputs
-		DDRB |= _BV(PRECHARGE_CTRL) | _BV(RJ45_LED1) | _BV(RJ45_LED2);
-		DDRC |= _BV(AIRMINUS_LSD);
-		DDRD |= _BV(LED1) | _BV(LED2) | _BV(LED3);
-}
+    // The following checks ensure that the hardware is in the correct initial
+    // state.
+    air_control_critical.air_p_status = !!gpio_get_pin(AIR_P_WELD_DETECT);
+    air_control_critical.air_n_status = !!gpio_get_pin(AIR_N_WELD_DETECT);
 
-void readAllInputs(void){
-		PCINT0_vect();
-		PCINT1_vect();
-		PCINT2_vect();
-}
+    if (air_control_critical.air_p_status) {
+        set_fault(AIR_FAULT_AIR_P_WELD);
+        rc = 1;
+        goto bail;
+    }
 
+    if (air_control_critical.air_n_status) {
+        set_fault(AIR_FAULT_AIR_N_WELD);
+        rc = 1;
+        goto bail;
+    }
 
-void checkBMSPowerStagePlausibility (void) { // tested and working. needs some type of delay as to when the check is made so it's not too soon
-		if ( (bit_is_set(gFlag, FLAG_BMS_STATUS) && bit_is_set(sFlag, FLAG_SS_TSCONN) && bit_is_clear(sFlag, FLAG_SS_BMS))
-		|| (bit_is_clear(gFlag, FLAG_BMS_STATUS) && bit_is_set(sFlag, FLAG_SS_BMS)) ) {
-			msgCritical[MSG_INDEX_ERROR_CODE] = FAULT_CODE_BMS_IMPLAUSIBILITY;
-		}
-}
+    if (!gpio_get_pin(SS_TSMS)) {
+        // SS_TSMS should start high
+        air_control_critical.ss_tsms = true;
+        set_fault(AIR_FAULT_SHUTDOWN_IMPLAUSIBILITY);
+        rc = 1;
+        goto bail;
+    }
 
-void checkIMDPowerStagePlausibility (void) { // tested and working
-		if ( (bit_is_set(gFlag, FLAG_IMD_STATUS) && bit_is_set(sFlag, FLAG_SS_BMS) && bit_is_clear(sFlag, FLAG_SS_IMD))
-		|| (bit_is_clear(gFlag, FLAG_IMD_STATUS) && bit_is_set(sFlag, FLAG_SS_IMD)) ) {
-			msgCritical[MSG_INDEX_ERROR_CODE] = FAULT_CODE_IMD_IMPLAUSIBILITY;
-		}
-}
+    if (!gpio_get_pin(IMD_SENSE)) {
+        // IMD_SENSE pin should start high
+        air_control_critical.imd_status = false;
+        set_fault(AIR_FAULT_IMD_STATUS);
+        rc = 1;
+        goto bail;
+    } else {
+        air_control_critical.imd_status = true;
+    }
 
-void checkAIRPLUS (void) {
-		if ( bit_is_set(gFlag, FLAG_AIRPLUS_WELD_DET) && bit_is_clear(gFlag, FLAG_TSMS_STATUS) ) {
-			msgCritical[MSG_INDEX_ERROR_CODE] = FAULT_CODE_AIRPLUS_WELD;
-		} else if ( bit_is_clear(gFlag, FLAG_AIRPLUS_WELD_DET) && bit_is_set(gFlag, FLAG_TSMS_STATUS) ) {
-			msgCritical[MSG_INDEX_ERROR_CODE] = FAULT_CODE_AIRPLUS_CONTROL_LOSS;
-		}
-}
+    // TODO implement hitl test
+    if (!gpio_get_pin(BMS_SENSE)) {
+        // IMD_SENSE pin should start high
```

Comment:
I think this comment was just copied from the previous block of code and forgot to be changed. Shouldn't it be the BMS_SENSE?

Reply:
Good catch